### PR TITLE
Add basic aggregate support

### PR DESCRIPTION
**Pull request by rlwww**

### CODE REVIEW

File: nemo/src/model/types/error.rs [hunk 1]
```diff
@@ -28,6 +28,9 @@ impl From<InvalidRuleTermConversion> for ReadingError {
 /// Errors that can occur during type checking
 #[derive(Error, Debug)]
 pub enum TypeError {
+    /// Non-numerical aggregate input type
+    #[error("Aggregate operation \"{0:?}\" on input variable \"{1}\" requires a numerical input type, but was input type was \"{2}\".")]
```
Comment (mmarx) [inline nemo/src/model/types/error.rs:32]: `but was input type was` ;)

File: nemo/src/model/chase_model/aggregate.rs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    pub(crate) aggregate_operation: AggregateOperation,
+    pub(crate) logical_aggregate_operation: LogicalAggregateOperation,
+
+    pub(crate) variable_identifiers: Vec<Identifier>,
+    /// Name of placeholder variable for the aggregate, which will get used in the corresponding head atom.
+    pub(crate) output_variable: Identifier,
+}
+
+impl ChaseAggregate {
+    #[allow(missing_docs)]
```
Comment (mmarx) [inline nemo/src/model/chase_model/aggregate.rs:17]: Why no docs here?

**Review by mmarx — CHANGES_REQUESTED** [review]
Looks good, just a few typos and some questions.

File: nemo-physical/src/aggregates/processors/processor.rs [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+/// Allows aggregation of multiple rows (all with the same group-by values) to produce a single aggregate value.
+pub trait AggregateGroupProcessor<A>
+where
+    A: Aggregate,
+{
+    /// Processes a row of the aggregated input column and updates the internal state.
+    fn write_aggregate_input_value(&mut self, value: A);
+
+    /// Returns the resulting aggregated value of all the processed input values.
+    fn finish(&self) -> Option<StorageValueT>;
```
Comment (monsterkrampe) [inline nemo-physical/src/aggregates/processors/processor.rs:28]: General Question: Shouldn't the aggregates work with `DataValueT`, or put differently: Do we not want to support aggregates on strings?

**Review by mmarx — APPROVED** [review]

**Review by monsterkrampe — APPROVED** [review]

File: nemo-physical/src/aggregates/processors/processor.rs (hunk 3 above)
Comment (monsterkrampe) [reply nemo-physical/src/aggregates/processors/processor.rs:28]: I'm aware that it's not easy to talk about `DataValueT` in a `TrieScan`. Intuitively, I feel that aggregation should happen on a higher level (but performance will suffer from that). Anyway, for now it should be fine :)

**Review by aannleax — CHANGES_REQUESTED** [review]
Very nice work :)

**Pull request merged**